### PR TITLE
React to Logging API changes

### PR DESCRIPTION
**Pull request by BrennanConroy**
https://github.com/aspnet/Logging/issues/331

### CODE REVIEW

File: src/Microsoft.AspNet.Hosting/Internal/HostingLoggerExtensions.cs [hunk 1]
```diff
@@ -6,6 +6,7 @@
 using System.Collections.Generic;
 using Microsoft.AspNet.Http;
 using Microsoft.Extensions.Logging;
+using System.Collections;
```
Comment (BrennanConroy) [inline src/Microsoft.AspNet.Hosting/Internal/HostingLoggerExtensions.cs]: fix

File: src/Microsoft.AspNet.Hosting/Internal/HostingLoggerExtensions.cs [hunk 2]
```diff
@@ -108,29 +133,66 @@ public override string ToString()
...
             private readonly HttpRequest _request;
 
             private string _cachedToString;
-            private IEnumerable<KeyValuePair<string, object>> _cachedGetValues;
+
+            public int Count
+            {
+                get
+                {
+                    return 9;
```
Comment (JunTaoLuo) [inline src/Microsoft.AspNet.Hosting/Internal/HostingLoggerExtensions.cs]: This doesn't look right. Magic numbers and hard coding indexers sounds like a bad idea. Why is this needed? cc @Tratcher @loudej

File: src/Microsoft.AspNet.Hosting/Internal/HostingLoggerExtensions.cs [hunk 3]
```diff
@@ -108,29 +133,66 @@ public override string ToString()
...
 
-                return _cachedGetValues;
+            IEnumerator IEnumerable.GetEnumerator()
+            {
+                return GetEnumerator();
             }
         }
 
-        private class HostingRequestStarting : ILogValues
+        private class HostingRequestStarting : IReadOnlyList<KeyValuePair<string, object>>
```
Comment (JunTaoLuo) [inline src/Microsoft.AspNet.Hosting/Internal/HostingLoggerExtensions.cs]: Is there a reason to implement `IReadOnlyList` if we provide a message formatter already?

File: src/Microsoft.AspNetCore.Hosting/Internal/HostingLoggerExtensions.cs [hunk 4]
```diff
@@ -108,29 +133,66 @@ public override string ToString()
...
 
-                return _cachedGetValues;
+            IEnumerator IEnumerable.GetEnumerator()
+            {
+                return GetEnumerator();
             }
         }
 
-        private class HostingRequestStarting : ILogValues
+        private class HostingRequestStarting : IReadOnlyList<KeyValuePair<string, object>>
```
Comment (nblumhardt) [inline src/Microsoft.AspNetCore.Hosting/Internal/HostingLoggerExtensions.cs]: Per #592 - `HostingRequestFinished` below requires updating as well.
